core/appconfig/config_test.go

```diff
@@ -119,3 +123,103 @@ modules:
 `))
 	expectContainerErrorContains(t, opt, "module should have ModuleDescriptor.go_import specified")
 }
+
+//
+// Test Module Initialization Logic
+//
```

Comment:
I needed to move this here because depinject complained that it was in an `internal` package before.